be/src/exprs/agg/percentile_cont.h

```diff
@@ -0,0 +1,172 @@
+
+template <PrimitiveType PT, typename = guard::Guard>
+struct PercentileContState {
+    using CppType = RunTimeCppType<PT>;
+    void update(CppType item) { items.emplace_back(item); }
+    void sort() { std::stable_sort(std::begin(items), std::end(items)); }
+
+    std::vector<CppType> items;
+    double rate = 0.0;
+};
+
+template <PrimitiveType PT>
+class PercentileContAggregateFunction final
+        : public AggregateFunctionBatchHelper<PercentileContState<PT>, PercentileContAggregateFunction<PT>> {
+public:
+    using InputCppType = RunTimeCppType<PT>;
+    using InputColumnType = RunTimeColumnType<PT>;
+    static constexpr auto ResultPT = PercentileResultPT<PT>;
+    using ResultType = RunTimeCppType<ResultPT>;
+    using ResultColumnType = RunTimeColumnType<ResultPT>;
+
+    void update(FunctionContext* ctx, const Column** columns, AggDataPtr __restrict state,
+                size_t row_num) const override {
+        const auto& column = down_cast<const InputColumnType&>(*columns[0]);
+        this->data(state).update(column.get_data()[row_num]);
+
+        if (ctx->get_num_args() == 2) {
+            const auto* rate = down_cast<const ConstColumn*>(columns[1]);
+            this->data(state).rate = rate->get(row_num).get_double();
+        }
+    }
+
+    void merge(FunctionContext* ctx, const Column* column, AggDataPtr __restrict state, size_t row_num) const override {
+        DCHECK(column->is_binary());
+
+        // Slice is a simple structure containing a pointer into some external storage and a size.
+        const Slice slice = column->get(row_num).get_slice();
+        double rate = *reinterpret_cast<double*>(slice.data);
+        size_t items_size = *reinterpret_cast<size_t*>(slice.data + sizeof(double));
+        auto data_ptr = slice.data + sizeof(double) + sizeof(size_t);
+
+        vector<InputCppType> res;
+        vector<InputCppType>& vec = this->data(state).items;
+        res.resize(vec.size() + items_size);
+
+        int i = 0, j = 0;
+        while (i < vec.size() || j < items_size) {
+            if (i == vec.size() && j < items_size) {
+                memcpy(res.data() + i + j, data_ptr + j * sizeof(InputCppType),
+                       (items_size - j) * sizeof(InputCppType));
+                break;
+            } else if (j == items_size && i < vec.size()) {
+                memcpy(res.data() + i + j, vec.data() + i, (vec.size() - i) * sizeof(InputCppType));
+                break;
+            } else if (i < vec.size() && j < items_size &&
+                       vec[i] > *reinterpret_cast<InputCppType*>(data_ptr + j * sizeof(InputCppType))) {
+                res[i + j] = *reinterpret_cast<InputCppType*>(data_ptr + j * sizeof(InputCppType));
+                j++;
+            } else if (i < vec.size()) {
+                res[i + j] = vec[i];
+                i++;
+            } else
+                break;
+        }
+        this->data(state).items = res;
```

Comment:
how about 
```suggestion
        this->data(state).items = std::move(res);
```
which can avoid memory copy

Reply:
fixed, thanks !